presto-orc/src/main/java/com/facebook/presto/orc/checkpoint/Checkpoints.java

```diff
@@ -73,65 +76,44 @@ private Checkpoints()
+            int sequence = 0;
+            ColumnEncoding columnEncoding = columnEncodings.get(column);
             OrcTypeKind columnType = columnTypes.get(column).getOrcTypeKind();
             Set<StreamKind> availableStreams = streamKinds.get(column);
+            Map<Integer, List<RowGroupIndex>> indexesForColumn = columnIndexes.get(column);
 
-            ColumnPositionsList columnPositionsList = new ColumnPositionsList(column, columnType, positionsList);
-            switch (columnType) {
-                case BOOLEAN:
-                    checkpoints.putAll(getBooleanColumnCheckpoints(column, compressed, availableStreams, columnPositionsList));
-                    break;
-                case BYTE:
-                    checkpoints.putAll(getByteColumnCheckpoints(column, compressed, availableStreams, columnPositionsList));
-                    break;
-                case SHORT:
-                case INT:
-                case LONG:
-                case DATE:
-                    checkpoints.putAll(getLongColumnCheckpoints(column, columnEncoding, compressed, availableStreams, columnPositionsList));
-                    break;
-                case FLOAT:
-                    checkpoints.putAll(getFloatColumnCheckpoints(column, compressed, availableStreams, columnPositionsList));
-                    break;
-                case DOUBLE:
-                    checkpoints.putAll(getDoubleColumnCheckpoints(column, compressed, availableStreams, columnPositionsList));
-                    break;
-                case TIMESTAMP:
-                    checkpoints.putAll(getTimestampColumnCheckpoints(column, columnEncoding, compressed, availableStreams, columnPositionsList));
-                    break;
-                case BINARY:
-                case STRING:
-                case VARCHAR:
-                case CHAR:
-                    checkpoints.putAll(getSliceColumnCheckpoints(column, columnEncoding, compressed, availableStreams, columnPositionsList));
-                    break;
-                case LIST:
-                case MAP:
-                    checkpoints.putAll(getListOrMapColumnCheckpoints(column, columnEncoding, compressed, availableStreams, columnPositionsList));
-                    break;
-                case STRUCT:
-                    checkpoints.putAll(getStructColumnCheckpoints(column, compressed, availableStreams, columnPositionsList));
-                    break;
-                case DECIMAL:
-                    checkpoints.putAll(getDecimalColumnCheckpoints(column, columnEncoding, compressed, availableStreams, columnPositionsList));
-                    break;
-                case UNION:
-                    throw new IllegalArgumentException("Unsupported column type " + columnType);
+            if (indexesForColumn == null) {
+                // This can happen if no data was written for the column, e.g. the key in a flat map
```

Comment:
This is an artifact of the way flat maps were written.  If the map wasn't flattened it would have a column for keys and a column for values.  If the map is flattened it still has those two columns plus a bunch of sequences, it's just that those 2 columns don't have any data, just statistics.